tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -93,13 +135,154 @@ public async Task<long> UploadRepositoryChangesAsync()
         var localCommits = gitOutput.Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
         if (localCommits.Length == 0)
         {
+            Log.Debug("ITR: Local commits not found. (since 1 month ago)");
             return 0;
         }
 
+        Log.Debug<int>("ITR: Local commits = {count}", localCommits.Length);
         var remoteCommitsData = await SearchCommitAsync(localCommits).ConfigureAwait(false);
         return await SendObjectsPackFileAsync(localCommits[0], remoteCommitsData).ConfigureAwait(false);
     }
 
+    public async Task<SettingsResponse> GetSettingsAsync(bool skipFrameworkInfo = false)
+    {
+        Log.Debug("ITR: Getting settings...");
+        var framework = FrameworkDescription.Instance;
+        var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
+        var branchName = await _getBranchNameTask.ConfigureAwait(false);
+        var currentSha = await _getShaTask.ConfigureAwait(false);
+        if (currentSha is null)
+        {
+            Log.Warning("ITR: 'git rev-parse HEAD' command is null");
+            return default;
+        }
+
+        currentSha = currentSha.Replace("\n", string.Empty);
+
+        var query = new DataEnvelope<Data<SettingsQuery>>(
+            new Data<SettingsQuery>(
+                currentSha,
+                SettingsType,
+                new SettingsQuery(
+                    _serviceName,
+                    _environment,
+                    repository,
+                    branchName,
+                    currentSha,
+                    new TestsConfigurations(
+                        framework.OSPlatform,
+                        Environment.OSVersion.VersionString,
+                        framework.OSArchitecture,
+                        skipFrameworkInfo ? null : framework.Name,
+                        skipFrameworkInfo ? null : framework.ProductVersion,
+                        skipFrameworkInfo ? null : framework.ProcessArchitecture))),
+            default);
+        var jsonQuery = JsonConvert.SerializeObject(query, SerializerSettings);
+        var jsonQueryBytes = Encoding.UTF8.GetBytes(jsonQuery);
+        Log.Debug("ITR: JSON RQ = {json}", jsonQuery);
+
+        return await WithRetries(InternalGetSettingsAsync, jsonQueryBytes, MaxRetries).ConfigureAwait(false);
+
+        async Task<SettingsResponse> InternalGetSettingsAsync(byte[] state, bool finalTry)
+        {
+            var request = _apiRequestFactory.Create(_settingsUrl);
+            request.AddHeader(ApiKeyHeader, _settings.ApiKey);
+            request.AddHeader(ApplicationKeyHeader, _settings.ApplicationKey);
+            request.AddHeader(HttpHeaderNames.TraceId, _id);
+            request.AddHeader(HttpHeaderNames.ParentId, _id);
+            Log.Debug("ITR: Getting settings from: {url}", _settingsUrl.ToString());
+            var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
```

Comment:
```suggestion
            using var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
```

Reply:
done in [6bc5a57](https://github.com/DataDog/dd-trace-dotnet/pull/3075/commits/6bc5a574614dc13585e91dedf1c8bd721bd4603a)